tests/test_bpc_redcap_export_mapping.py

```diff
@@ -113,3 +116,51 @@ def test_get_mapping_data_calls_dd_if_use_grs_is_false(mock_syn):
 def test_that_parse_drug_mappings(input_mapping, var_names, output_mapping):
     result = bpc_export.parse_drug_mappings(mapping=input_mapping, var_names=var_names)
     assert result == output_mapping
+
+
+@pytest.mark.parametrize(
+    "input_data, oncotree_dict, expected_warning",
+    [
+        (
+            pd.DataFrame(
+                dict(
+                    ONCOTREE_CODE=["Renal Cell Carcinoma", "Renal Clear Cell Carcinoma"]
+                )
+            ),
+            {
+                "RCC": {"CANCER_TYPE": "Renal Cell Carcinoma"},
+                "OVARY": {"CANCER_TYPE": "Ovarian Cancer"},
+            },
+            "There are invalid values in ONCOTREE_CODE column in the clinical df: ['Renal Cell Carcinoma', 'Renal Clear Cell Carcinoma']",
+        ),
+        (
+            pd.DataFrame(dict(ONCOTREE_CODE=["Renal Cell Carcinoma", "RCC"])),
+            {
+                "RCC": {"CANCER_TYPE": "Renal Cell Carcinoma"},
+                "OVARY": {"CANCER_TYPE": "Ovarian Cancer"},
+            },
+            "There are invalid values in ONCOTREE_CODE column in the clinical df: ['Renal Cell Carcinoma']",
+        ),
+    ],
+    ids=["all_invalid", "some_invalid"],
+)
+def test_that_check_oncotree_codes_gives_expected_warning_when_invalid_codes(
+    caplog, input_data, oncotree_dict, expected_warning
```

Comment:
TIL: caplog! Is that a pytest thing?

Reply:
Yup it is:https://docs.pytest.org/en/6.2.x/logging.html#caplog-fixture. I think we used this in recover, main genie and I saw it in synapse python client.